sys/arm64/include/atomic.h

```diff
@@ -71,6 +71,12 @@ extern bool lse_supported;
 #define	_ATOMIC_LSE_SUPPORTED	0
 #endif
 
+#ifndef __CHERI_PURE_CAPABILITY__
+#define ATOMIC_PTR_CONST	"r"
```

Comment:
Being more verbose is probably too long, but I read this as "const" rather than "constraint" at first.

Reply:
This should have an underscore prefix though to be in keeping with the other macros, no?

Reply:
Can we just use the __atomic_* builtins instead? Should be something that can be upstreamed and should generate better code since there aren't any more "memory" clobbers.

Reply:
We do that on RISC-V in CheriBSD now.  We haven't yet gotten that upstreamed as mjg@ wants to do it a bit differently than we did.

Reply:
The arm64 atomic functions handle both traditional LL/SC atomics from armv8.0, and the Large System Extension in armv8.1. Would the buildins allow us to handle these?